test-packages/test-utils/src/utils/download-ui5-resources.ts

```diff
@@ -0,0 +1,78 @@
+import { TestModelVersion } from "../../api";
+import { reduce, keys, map } from "lodash";
+import { resolve } from "path";
+import { writeFile, mkdirs, pathExists } from "fs-extra";
+import fetch from "node-fetch";
+
+async function getLibs(version: TestModelVersion): Promise<string[]> {
+  // The metadata.json seems to have been added only very recently :(
+  let versionInURL: string = version;
+  if (versionInURL !== "1.76.0") {
+    versionInURL = "1.76.0";
+  }
+  const url = `https://unpkg.com/@sapui5/distribution-metadata@${versionInURL}/metadata.json`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    console.log(`error fetching from ${url}`);
+    return [];
+  }
+  const fileContent = await response.text();
+  const librariesMetadata = JSON.parse(fileContent);
+  return keys(librariesMetadata.libraries);
+}
+
+export async function addUi5Resources(
+  version: TestModelVersion,
+  folder: string
+): Promise<void> {
+  // CDN libraries (example URL):
+  // https://sapui5-sapui5.dispatcher.us1.hana.ondemand.com/test-resources/sap/m/designtime/api.json
+  // Older versions:
+  // https://sapui5.hana.ondemand.com/1.71.14/test-resources/sap/m/designtime/api.json
+  const baseUrl = `https://sapui5.hana.ondemand.com/${version}/test-resources/`;
+  const libs = await getLibs(version);
+  const nameToFile = reduce(
```

Comment:
This is **subjective**:

Because the size of the input and output is the same
this **reduce** is not much more than a glorified forEach.

What do you think of combining Map and ZipObject
- https://lodash.com/docs/4.17.15#zipObject

To obtain he same result? 
would it be clearer?

Reply:
Yes, I'll change it